src/components/DeviceListTable.tsx

```diff
@@ -0,0 +1,117 @@
+import { FC } from "react";
+import { MainTable } from "@canonical/react-components";
+import { LxdInstance } from "types/instance";
+import { getInstanceDevices, getProfileDevices } from "util/helpers";
+import { isRootDisk } from "util/instanceValidation";
+import { FormDevice } from "util/formDevices";
+import ResourceLink from "components/ResourceLink";
+import { isOtherDevice } from "util/devices";
+import InstanceOverviewDeviceDetail from "../pages/instances/InstanceOverviewDeviceDetail";
+import { LxdProfile } from "types/profile";
+import { useParams } from "react-router-dom";
+import { LxdDeviceValue } from "types/device";
+
+interface Props {
+  instance?: LxdInstance;
+  profile?: LxdProfile;
```

Comment:
This is not a good interface, because both props are mutual exclusive. This leads to a lot of branching in this component. I think better would be to make this component agnostic of instances and profiles. So we pass in only two generic props: `devices` as an array and the `configBaseUrl` as a string.